crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -71,44 +77,75 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
                 }
                 write!(f, [first.format()])?;
             }
-            SuiteKind::Class if is_docstring(first) => {
-                if !comments.has_leading_comments(first) && lines_before(first.start(), source) > 1
-                {
-                    // Allow up to one empty line before a class docstring, e.g., this is
-                    // stable formatting:
+            SuiteKind::Function => {
+                if let Some(constant) = get_docstring(first, &f.context().locator()) {
+                    write!(
+                        f,
+                        [
+                            // We format the expression, but the statement carries the comments
+                            leading_comments(comments.leading_comments(first)),
+                            constant
+                                .format()
+                                .with_options(ExprConstantLayout::String(StringLayout::DocString)),
+                            trailing_comments(comments.trailing_comments(first)),
+                        ]
+                    )?;
```

Comment:
Hmm, this will get tricky with suppression comments because I had to move the formatting call out of this match (to avoid having to test whether the expression is suppressed in multiple places)